packages/ui/tests/IntegrationProgress.spec.tsx

```diff
@@ -0,0 +1,58 @@
+import * as React from 'react';
+import { Matcher, MatcherOptions, render } from 'react-testing-library';
+import { IntegrationProgress } from '../src';
+
+export default describe('IntegrationProgress', () => {
+  // Component with a log link
+  const testComponent = (
+    <IntegrationProgress
+      value={'Deploying'}
+      currentStep={2}
+      totalSteps={4}
+      logUrl={'http://localhost:9000/'}
+      i18nLogUrlText={'View Logs'}
+    />
+  );
+
+  // Component without a log link
+  const testComponentNoLog = (
+    <IntegrationProgress value={'Assembling'} currentStep={1} totalSteps={4} />
+  );
+
+  it('Should show the progress value and steps', () => {
+    const { getByTestId } = render(testComponent);
+    expect(getByTestId('integration-progress-value')).toHaveTextContent(
+      'Deploying ( 2 / 4 )'
+    );
+  });
+  it('Should show the log link when supplied', () => {
+    const { getByTestId } = render(testComponent);
+    const logSpan = getLogSpan('deployment-log-link', getByTestId);
+    expect(logSpan).toBeDefined();
+    expect(logSpan).toHaveTextContent('View Logs');
+  });
+  it('Should show the progress value and steps', () => {
+    const { getByTestId } = render(testComponentNoLog);
+    expect(getByTestId('integration-progress-value')).toHaveTextContent(
+      'Assembling ( 1 / 4 )'
+    );
+  });
+  it('Should not show the log link if not supplied', () => {
+    const { getByTestId } = render(testComponentNoLog);
+    const logSpan = getLogSpan('deployment-log-link', getByTestId);
```

Comment:
Every `getXXX` has a [`queryXXX` counterpart](https://github.com/kentcdodds/react-testing-library#query-apis) that can be used to verify for the absence of an element.

You should be able to simplify this using a `queryByTestId` and then check for `null`

Reply:
ah, excellent, thanks!